### PR TITLE
[Custom Trainer] refactor normalization callbacks

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## Description

- Moves normalization callbacks into the trainer.

### CODE REVIEW

**Review by samet-akcay — COMMENTED** [review]
Thanks @ashwinvaidya17. Some minor comments.

File: src/anomalib/utils/callbacks/__init__.py [hunk 1]
```diff
@@ -16,24 +16,20 @@
...
 from .metrics_configuration import MetricsConfigurationCallback
-from .min_max_normalization import MinMaxNormalizationCallback
 from .model_loader import LoadModelCallback
 from .post_processing_configuration import PostProcessingConfigurationCallback
 from .tiler_configuration import TilerConfigurationCallback
 from .timer import TimerCallback
 from .visualizer import ImageVisualizerCallback, MetricVisualizerCallback
 
 __all__ = [
-    "CdfNormalizationCallback",
```
[the remark this answers is not in the document]
Comment (ashwinvaidya17) [reply src/anomalib/utils/callbacks/__init__.py:30]: I just thought that since we assign normalization callbacks in the trainer, I should make them part of the trainer's callbacks. But maybe I can keep them here and add a note to the docstring

File: src/anomalib/trainer/callbacks/normalization/cdf_normalization.py [hunk 2]
```diff
@@ -104,16 +103,16 @@ def on_predict_batch_end(
...
         """Collect the statistics of the normal training data.
 
         Create a trainer and use it to predict the anomaly maps and scores of the normal training data. Then
          estimate the distribution of anomaly scores for normal data at the image and pixel level by computing
          the mean and standard deviations. A dictionary containing the computed statistics is stored in self.stats.
         """
-        predictions = AnomalibTrainer(accelerator=trainer.accelerator, devices=trainer.num_devices).predict(
-            model=self._create_inference_model(pl_module), dataloaders=trainer.datamodule.train_dataloader()
-        )
+        predictions = anomalib.AnomalibTrainer(
```
[the remark this answers is not in the document]
Comment (ashwinvaidya17) [reply src/anomalib/trainer/callbacks/normalization/cdf_normalization.py:113]: 1. It looks a bit weird with `anomalib.AnomalibTrainer` just to avoid circular import. I'll try to find a better solution for this.
2. I guess the idea is to get normalization statistics from the training set instead of the validation set. Dick is probably the best person to address this but I did notice some difference in scores when I used the validation data. Plus the number of images in validation set might not be large enough to compute a reasonable statistics. We can't move gathering the stats to the training step as a trained model is not available yet to produce a decent anomaly map. But maybe the whole normalization requires refactor. 
3. 👍

File: src/anomalib/utils/callbacks/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (samet-akcay) [reply src/anomalib/utils/callbacks/__init__.py:30]: Just had a second thought. I think it would still be more organized if we keep the callbacks in one place. We might need to re-organize the relative imports to not be able to import these callbacks easily

https://github.com/openvinotoolkit/anomalib/blob/ed1d9fa5c7050a6edf79febd256d27b261a441ae/src/anomalib/utils/callbacks/__init__.py#L31

Thoughts?

**Review by samet-akcay — APPROVED** [review]
thanks! I think we could merge it now

**Comment by samet-akcay** [comment]
docs fail for some reason
